backend/src/auth/twofa/twofa.service.ts

```diff
@@ -0,0 +1,52 @@
+import { Injectable } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import { UserService } from '../../users/user/user.service';
+import * as crypto from 'crypto';
+import * as QRCode from 'qrcode';
+import { authenticator } from 'otplib';
+import { TwoFaDto } from './dto/twofa.dto';
+
+// we might want to consider encrypting/decrypting the secret
+// instead of saving it plaintext into the db
+//
+// import { scrypt, randomFill, createCipheriv } from 'crypto';
+// https://nodejs.org/api/crypto.html#class-cipher
+//
```

Comment:
I agree,  not sure if it's a little out of scope though?

Reply:
mjahshah, the eval sheet says to "verify that passwords are hashed"
and this is basically a password..
even though "hashing" would definitely not work here, since we need the actual value...

encrypting and decrypting it shouldn't be that much work tbh. but I would put it last on the list of things to do before Tuesday..